### DISCUSSION TITLE
[Solved] Power consumption refresh rate: 30 sec in the integration, 5 sec in the app

### DISCUSSION THREAD

**Discussion by ibennani** [petretiandrea/home-assistant-tapo-p100#391] [General]
Hello.
First of all, **a huge thanks to the developer** for this integration. It solved an [issue I have with the AppleTV](https://github.com/home-assistant/core/issues/76781). 

I noticed that the power consumption sensor is updated every 30 second, but just in this integration. The Tapo mobile app updates the information every 5 second. 

Is there a setting somewhere to change the power update to 5 seconds? Perhaps some file that can be edited?

**Comment by ibennani** [comment]
Solution: 

1. In the folder `/config/custom_components/tapo` , open the file `common_setup.py`.
2. Locate the line `SCAN_INTERVAL` and change the value at the end of the line from 30 to 5.
3. Save the file and restart Home Assistant.

Now the plug will report the power consumption every 5 seconds.

**Reply by petretiandrea** [reply]
Now the 1.7.1 supports advanced settings allowing to set refresh rate
